Source/Core/DiscIO/DiscExtractor.cpp

```diff
@@ -130,13 +130,15 @@ void ExportDirectory(const Volume& volume, const Partition& partition, const Fil
                      const std::string& export_folder,
                      const std::function<bool(const std::string& path)>& update_progress)
 {
-  File::CreateFullPath(export_folder + '/');
+  File::CreateFullPath(export_folder +
+                       (directory.IsDirectory() ? "/" + directory.GetName() + "/" : "/"));
```

Comment:
I would probably make sense to put this ternary in a local variable (perhaps even including the `export_folder` itself), rather than doing it both here and once for every iteration in the loop below.

Reply:
Changed. Generally speaking, do we want dolphin to be more CPU-bound or have a smaller memory footprint? I realize this is probably a non-issue here since this operation isn't time-sensitive and not particularly slow to begin with, but I can imagine similar situations elsewhere in the codebase.

Reply:
When it comes to the emulation core, being constrained by the CPU is way more common than being constrained by the RAM size, to the point where I can't remember a single time the latter has happened for anyone other than when custom texture prefetching is involved. When it comes to things other than the emulation core... Well, the RAM usage is even lower when not emulated, so using a bit of extra RAM should be no problem. Not that I believe keeping this string around actually increases the memory usage, as the alternative of reconstructing it every time still has to allocate the string for the duration of the lines where it's used.